src/Core/src/Credentials/PodIdentityProvider.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Core\Credentials;
+
+use AsyncAws\Core\Configuration;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Provides Credentials for running Pod Identity inside an EKS Cluster.
+ *
+ * @see https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html
+ */
+final class PodIdentityProvider implements CredentialProvider
```

Comment:
What's about reusing the existing `ContainerProvider` instead?

It looks like this is what the official AWS SDK does: https://github.com/aws/aws-sdk-php/blob/master/src/Credentials/EcsCredentialProvider.php

In that way, we could leverage the AuthToken in both full / relative URI 


Endpoint is one of:
- `http://169.254.170.2` . `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`
- pr `AWS_CONTAINER_CREDENTIALS_FULL_URI`

Auth Token is one of:
- `AWS_CONTAINER_AUTHORIZATION_TOKEN`
- content of `AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE`
- nothing

Reply:
I changed my implementation to reuse the `ContainerProvider` like the AWS SDK does 😄